### PR TITLE
merge VerifyTask and VerifyPipeline into VerifyResource

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit merges VerifyTask and VerifyPipeline into VerifyResource to reduce duplicate code.

/kind cleanup

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: VerifyTask and VerifyPipeline are now merged into 1 function VerifyResource, please update the usages if upgrade to the new release
```

### CODE REVIEW

**Comment by Yongxuanzhang** [comment]
/retest
https://storage.googleapis.com/tekton-prow/pr-logs/pull/tektoncd_pipeline/6724/pull-tekton-pipeline-alpha-integration-tests/1661852264114950144/build-log.txt
might be some issue with resolving the images:
```
         "step-show-user-2000" exited with code 1 (image: "docker.io/library/ubuntu@sha256:dfd64a3b4296d8c9b62aa3309984f8620b98d87e47492599ee20739e8eb54fbf"); for logs run: kubectl -n arendelle-s9bkm logs show-non-root-steps-run-jghvr-pod -c step-show-user-2000
```

File: pkg/trustedresources/verify.go [hunk 1]
```diff
@@ -91,61 +92,35 @@ func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Inter
...
-				logger := logging.FromContext(ctx)
-				warning := fmt.Errorf("failed to get matched policies: %w", err)
-				logger.Warnf(warning.Error())
-				return VerificationResult{VerificationResultType: VerificationWarn, Err: warning}
-			}
+	switch v := resource.(type) {
+	case *v1beta1.Task:
+		tm, signature, err := prepareObjectMeta(v.ObjectMeta)
+		if err != nil {
+			return VerificationResult{VerificationResultType: VerificationError, Err: err}
```
Comment (lbernick) [inline pkg/trustedresources/verify.go:99]: can this block be moved out of the switch statement? I think `metav1.Object` should have ObjectMeta.

Can you also link the TODO for supporting v1 task/pipeline verification?

File: pkg/trustedresources/verify.go [hunk 2]
```diff
@@ -63,19 +63,20 @@ type VerificationResult struct {
...
 }
 
-// VerifyTask verifies the signature and public key against task.
+// VerifyResource verifies the signature and public key against resource (v1beta1 task and pipeline).
 // VerificationResult is returned with different types for different cases:
 // 1) Return VerificationResult with VerificationSkip type, when no policies are found and no-match-policy is set to ignore
 // 2) Return VerificationResult with VerificationPass type when verification passed;
 // 3) Return VerificationResult with VerificationWarn type, when no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail. Err field is filled with the warning;
 // 4) Return VerificationResult with VerificationError type when no policies are found and no-match-policy is set to fail, the resource fails to pass matched enforce verification policy, or there are errors during verification. Err is filled with the err.
 // refSource contains the source information of the task.
```
Comment (lbernick) [inline pkg/trustedresources/verify.go:72]: can you update the reference to "task" here?

**Review by lbernick — APPROVED** [review]

**Comment by lbernick** [comment]
> /retest https://storage.googleapis.com/tekton-prow/pr-logs/pull/tektoncd_pipeline/6724/pull-tekton-pipeline-alpha-integration-tests/1661852264114950144/build-log.txt might be some issue with resolving the images:
> 
> ```
>          "step-show-user-2000" exited with code 1 (image: "docker.io/library/ubuntu@sha256:dfd64a3b4296d8c9b62aa3309984f8620b98d87e47492599ee20739e8eb54fbf"); for logs run: kubectl -n arendelle-s9bkm logs show-non-root-steps-run-jghvr-pod -c step-show-user-2000
> ```

Can you please open an issue tracking this flake?

**Comment by Yongxuanzhang** [comment]
> > /retest https://storage.googleapis.com/tekton-prow/pr-logs/pull/tektoncd_pipeline/6724/pull-tekton-pipeline-alpha-integration-tests/1661852264114950144/build-log.txt might be some issue with resolving the images:
> > ```
> >          "step-show-user-2000" exited with code 1 (image: "docker.io/library/ubuntu@sha256:dfd64a3b4296d8c9b62aa3309984f8620b98d87e47492599ee20739e8eb54fbf"); for logs run: kubectl -n arendelle-s9bkm logs show-non-root-steps-run-jghvr-pod -c step-show-user-2000
> > ```
> 
> Can you please open an issue tracking this flake?

I was thinking whether we should open an issue for this, since it doesn't seem to be an issue from tekton? I will open one just for record

**Comment by lbernick** [comment]
sorry Yongxuan I accidentally edited your comment instead of quote replying it!

**Comment by Yongxuanzhang** [comment]
> sorry Yongxuan I accidentally edited your comment instead of quote replying it!

Oh I see, thanks! I'm opening one now.

File: pkg/trustedresources/verify.go [hunk 3]
```diff
@@ -63,19 +63,21 @@ type VerificationResult struct {
...
 
-// VerifyTask verifies the signature and public key against task.
+// VerifyResource verifies the signature and public key against resource (v1beta1 task and pipeline).
 // VerificationResult is returned with different types for different cases:
 // 1) Return VerificationResult with VerificationSkip type, when no policies are found and no-match-policy is set to ignore
 // 2) Return VerificationResult with VerificationPass type when verification passed;
 // 3) Return VerificationResult with VerificationWarn type, when no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail. Err field is filled with the warning;
 // 4) Return VerificationResult with VerificationError type when no policies are found and no-match-policy is set to fail, the resource fails to pass matched enforce verification policy, or there are errors during verification. Err is filled with the err.
-// refSource contains the source information of the task.
-func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationpolicies []*v1alpha1.VerificationPolicy) VerificationResult {
```
Comment (JeromeJu) [inline pkg/trustedresources/verify.go:73]: Wondering if there is a way to reserve refactored functions like what we did for deprecated fields if that would be better? ie. func VerifyTask(..., task, ...) {VerifyResource(..., task, ...)} ?

Comment (Yongxuanzhang) [reply pkg/trustedresources/verify.go:73]: That's a good point, 🤔 Let me try that

Comment (Yongxuanzhang) [reply pkg/trustedresources/verify.go:73]: Added

**Comment by Yongxuanzhang** [comment]
/test pull-tekton-pipeline-go-coverage
network issue

**Review by JeromeJu — APPROVED** [review]

**Review by JeromeJu — APPROVED** [review]

**Pull request merged**